exec/java-exec/src/main/java/org/apache/drill/exec/planner/ParquetPartitionDescriptor.java

```diff
@@ -125,4 +117,16 @@ private String getBaseTableLocation() {
     final FormatSelection origSelection = (FormatSelection) scanRel.getDrillTable().getSelection();
     return origSelection.getSelection().selectionRoot;
   }
+
+  @Override
+  protected void createPartitionSublists() {
+    Set<String> fileLocations = ((ParquetGroupScan) scanRel.getGroupScan()).getFileSet();
+    List<PartitionLocation> locations = new LinkedList<>();
+    for (String file: fileLocations) {
+      locations.add(new DFSPartitionLocation(MAX_NESTED_SUBDIRS, getBaseTableLocation(), file));
```

Comment:
Long file names are an issue not just for partition pruning but metadata in general..that's what I was saying previously about FormatSelection.getAsFiles() and ParquetGroupScan.getFileSet() etc.  If we want to put the names into direct memory rather than heap, then a broader change is needed.   We should have a separate JIRA for that I think.